pkg/apis/core/validation/seed.go

```diff
@@ -341,6 +365,10 @@ func ValidateSeedSpecUpdate(newSeedSpec, oldSeedSpec *core.SeedSpec, fldPath *fi
 	}
 	// If oldSeedSpec doesn't have backup configured, we allow to add it; but not the vice versa.
 
+	if oldSeedSpec.DNS.Internal != nil && newSeedSpec.DNS.Internal == nil {
```

Comment:
Should we also validate that it's not allowed to change the domain itself?

Reply:
Yes, true. This is what we used to check with the help of https://github.com/gardener/gardener/blob/master/pkg/admissioncontroller/webhook/admission/internaldomainsecret :)

Reply:
Implemented with https://github.com/gardener/gardener/pull/12663/commits/1854fc5ff1f9bb316230debe1773c81d5b7f8e2d